server/controllers/role.js

```diff
@@ -0,0 +1,127 @@
+const Role = require('../models/role');
+
+module.exports = {
+
+  // Create a role
+  create: (req, res) => {
+    const role = new Role();
+
+    role.title = req.body.title;
+    role.permission = req.body.permission;
+
+    role.save(function(err) {
+      if (err) {
+        if (err.code === 11000) {
+          res.json({
+            message: 'Please provide a unique title',
```

Comment:
Will this return a 200 status code? It would be best to return an error code with the message.